File: src/main/java/io/shcm/shsupercm/fabric/stonecutter/cutter/FileCutter.java
package io.shcm.shsupercm.fabric.stonecutter.cutter;

import io.shcm.shsupercm.fabric.stonecutter.processor.CommentProcessor;

import java.io.File;
import java.io.Reader;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.StandardOpenOption;

public class FileCutter {
    private final File file;
    private final StonecutterTask stonecutter;

    public FileCutter(File file, StonecutterTask stonecutter) {
        this.file = file;
        this.stonecutter = stonecutter;
    }

    @SuppressWarnings("ResultOfMethodCallIgnored")
    public void write(File outputFile) throws Exception {
        StringBuilder transformedContents;
        try (Reader oldContents = Files.newBufferedReader(file.toPath(), StandardCharsets.ISO_8859_1)) {
            transformedContents = CommentProcessor.process(oldContents, stonecutter.processor());
            stonecutter.tokenRemapper().apply(file, transformedContents);
        }

        outputFile.delete();
        Files.writeString(outputFile.toPath(), transformedContents, StandardCharsets.ISO_8859_1, StandardOpenOption.CREATE);
    }
}

File: src/main/kotlin/dev/kikugie/stonecutter/cutter/FileCutter.kt
package dev.kikugie.stonecutter.cutter

import dev.kikugie.stonecutter.processor.CommentProcessor
import java.nio.charset.StandardCharsets
import java.nio.file.Path
import java.nio.file.StandardOpenOption
import kotlin.io.path.bufferedReader
import kotlin.io.path.writeText

class FileCutter(
    val file: Path,
    val stonecutter: StonecutterTask
) {
    fun write(output: Path) {
        val transformed = try {
            CommentProcessor.process(file.bufferedReader(StandardCharsets.ISO_8859_1), stonecutter.processor)
        } catch (e: Exception) {
            throw RuntimeException("Failed processing file $file:\n${e.message}", e)
        }
        output.writeText(transformed, StandardCharsets.ISO_8859_1, StandardOpenOption.CREATE)
    }
}